pkg/entrypoint/entrypointer.go

```diff
@@ -110,6 +112,8 @@ type Entrypointer struct {
 	StepMetadataDir string
 	// SpireWorkloadAPI connects to spire and does obtains SVID based on taskrun
 	SpireWorkloadAPI spire.EntrypointerAPIClient
+	// StepName is the name of the step
+	StepName string
```

Comment:
I don't think you need this. IIUC this is used in 2 places:

1. For the result path, which can be replaced with `filepath.Join(e.StepMetadataDir, "results")`
2. For the result names, which is unneeded because the termination message is already attached to the matching container anyway - the controller can derive this information from the container.

Reply:
agreed!